presto-main/src/main/java/com/facebook/presto/operator/scalar/MapFilterFunction.java

```diff
@@ -132,18 +142,21 @@ private static MethodHandle generateFilter(Type keyType, Type valueType, Type ma
                 type(Object.class));
         definition.declareDefaultConstructor(a(PRIVATE));
 
+        Parameter state = arg("state", Object.class);
         Parameter block = arg("block", Block.class);
         Parameter function = arg("function", BinaryFunctionInterface.class);
         MethodDefinition method = definition.declareMethod(
                 a(PUBLIC, STATIC),
                 "filter",
                 type(Block.class),
-                ImmutableList.of(block, function));
+                ImmutableList.of(state, block, function));
 
         BytecodeBlock body = method.getBody();
         Scope scope = method.getScope();
         Variable positionCount = scope.declareVariable(int.class, "positionCount");
         Variable position = scope.declareVariable(int.class, "position");
+        Variable pageBuilder = scope.declareVariable(PageBuilder.class, "pageBuilder");
+        Variable mapBlockBuilder = scope.declareVariable(BlockBuilder.class, "mapBlockBuilder");
         Variable blockBuilder = scope.declareVariable(BlockBuilder.class, "blockBuilder");
```

Comment:
I think usually you like to use `singleMapBlockWriter` here as the variable name :)